src/orquestra/opt/optimizers/scipy_optimizer.py

```diff
@@ -14,6 +14,77 @@
 )
 from ..history.recorder import RecorderFactory
 from ..history.recorder import recorder as _recorder
+from .pso.continuous_pso_optimizer import _get_bounds_like_array
+
+
+class _CostFunctionWithBestValueType(type):
+    def __instancecheck__(cls, obj: object) -> bool:
+        if (
+            isinstance(obj, Callable)
+            and hasattr(obj, "best_value")
+            and hasattr(obj, "best_params")
+        ):
+            return True
+        return False
+
+
+class _CostFunctionWithBestValue(metaclass=_CostFunctionWithBestValueType):
+    best_value: float = np.inf
+    best_params: np.ndarray = np.empty(1)
+
+    def __init__(
+        self,
+        cost_function: Union[CallableWithGradient, Callable],
+        constraints: Sequence[Dict[str, Callable]],
+        bounds: Union[
+            scipy.optimize.Bounds,
+            Sequence[Tuple[float, float]],
+            None,
+        ] = None,
+    ) -> None:
+        self.cost_function = cost_function
+        # Inherit all attributes from the cost function
+        for attr in dir(cost_function):
+            if not attr.startswith("__"):
+                setattr(self, attr, getattr(cost_function, attr))
```

Comment:
When making a composition, it's usually best to get the attributes from the wrapped objects rather than just copying them over. This is because you can easily end up accidentally importing attributes that might change your execution. That is, your object won't be very consistent between instances. I would suggest that you really look to ensure there are no alternatives before doing this sort of thing.

Reply:
Yes, this can be a bit dangerous. I will perform a manual check for the time being so that there are no collisions with the methods and other attribute names.